lib/spack/spack/main.py

```diff
@@ -341,6 +347,10 @@ def setup_main_options(args):
     tty.set_stacktrace(args.stacktrace)
     spack.debug = args.debug
 
+    # override lock configuration if passed on command line
+    if args.locks is not None:
+        spack.locks = args.locks
```

Comment:
I wonder if we should expose this option to the command line at all. If we enable/disable locks, it should be for all the users of a particular Spack instance (and we should leave them no way to work around this setting without modifying the configuration file).